lib/class-oa2-authenticator.php

```diff
@@ -34,14 +34,13 @@ public function __construct() {
    * @return WP_User|null|WP_Error Authenticated user on success, null if no OAuth data supplied, error otherwise
    */
   public function authenticate( $user ) {
-	if ( !empty( $user ) ) {
-	  return $user;
-	}
-
 	// Check if token is in request
 	$bearer_token = OA2_Header_Helper::get_authorization_bearer();
 	if ( empty( $bearer_token ) ) {
 	  $this->auth_status = $bearer_token;
+      if ( !empty( $user ) ) {
```

Comment:
Yes, that makes sense. However, my problem is: Why are you using the token in the same browser that you use to log in? Usually Authorization Code grant from OAuth 2.0 is used for third-party apps. Are you using the OAuth 2.0 token with an app that is running in the browser?